Data/ByteString/Lazy.hs

```diff
@@ -525,6 +526,49 @@ minimum (Chunk c cs) = foldlChunks (\n c' -> n `min` S.minimum c')
                                      (S.minimum c) cs
 {-# INLINE minimum #-}
 
+-- | /O(n)/ 'compareLength' compares the length of a 'ByteString' 
+-- to an 'Int64'   
+compareLength :: ByteString -> Int64 -> Ordering
+compareLength _ toCmp | toCmp < 0 = GT
+compareLength Empty toCmp         = compare 0 toCmp
+compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - fromIntegral (S.length c))
+{-# INLINE compareLength #-}
+
+{-# RULES 
+"ByteString.Lazy compareN/length -> compareLength" [~1] forall t n.
+  compare (length t) n = compareLength t n
+  #-}
+
+{-# RULES 
+"ByteString.Lazy ==N/length -> compareLength/==EQ" [~1] forall t n.
+   length t == n = compareLength t n == EQ
+  #-}
```

Comment:
Similarly to my comment above, shouldn't we also have rules for `n == length t`? (This also applies to the rules below)

Reply:
done